packages/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -59,6 +72,86 @@ export const useMenu = (props: MenuProps, ref: React.Ref<HTMLElement>, defaultPr
     },
     [setOpen],
   );
+  state.on.length === 0 && state.on.push('click');
```

Comment:
Do I correctly get that `state.on` is actually `props.on`? In this case we can't use `.push()` there as it will mutate user's input.

```suggestion
  if(state.on.length === 0) {
    state.on = ['click'];
  }
```

I also suggest to add a comment why it is required: as I understood it applies some default value